### ISSUE TITLE
Cannot unset `background-image` from Heading Background in mobile theme

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#2143]
Cannot unset the `background-image` property in the mobile theme editor.

Steps:
1. Clone the iPhone theme.
2. Select the **Heading Background** widget.
3. In **Properties > Background**, unset the `background-image` property.
   - _Notice that there are no changes to the widget._
4. Click to a different widget, then back to Heading Background.
   - _Notice that the value in `background-image` has reappeared._

@JonFerraiolo

**Comment by JonFerraiolo** [comment]
@billreed63 - feel free to assign to me if the background properties logic is hard to fathom.

**Comment by JonFerraiolo** [comment]
@billreed63 - hope it's OK, I snatched this issue. I figure you will be busy for awhile with the hotmodify issue and I know  the background code, which is quite complicated

**Comment by JonFerraiolo** [comment]
-> back to @billreed63. 

In tracing things in the debugger, changing background-image value (which was "-webkit-gradient..." to empty string goes through all of the correct machinery in Cascade.js, then ThemeEditor.js, and finally ModifyRule() in Context.js to hot-modify the rule. That's all goodness.

But the problem (I believe) is Context.js:2423:

```
if (cleaned[i][name] && cleaned[i][name] !== '') { 
```

I think the problem here is that things need to work differently with delta themes. In the old world, we could remove a property by simply removing the property declaration from the rule (which is done in lines above 2423). But with delta themes, if we remove the rule, we are removing it from the delta, which (in effect) causes the original rule from the original theme to be restored because there is no override rule in the delta theme anymore.

Unfortunately, I think the only solution to this is that we need to put in special-case logic for the theme editor (and delta themes) so that an empty string value has to stuff in the "Initial value" for the property (see Initial value column at http://www.w3.org/TR/CSS21/propidx.html). In order to do that, we will probably need to include an initial value table for the various CSS properties that we support (e.g., {'background-color':'transparent','background-image':'none',...}). If we go that route, I would nominate either adding a CSS properties table to davinci/ve/CSSUtils.js or (probably better) adding another JavaScript file that includes the initial values table.

**Comment by JonFerraiolo** [comment]
Raising to Prio High because this seems like a pretty fundamental problem with our delta themes implementation.

**Comment by billreed63** [comment]
This is a bigger UI issue as described in #2328
moving to M7

**Comment by billreed63** [comment]
I disagree with @JonFerraiolo, I think that the current behavior is the correct behavior. If the user changes there mind and removes a value, we should just remove that one property from the delta rule and let it default to the base themes rule. If the user wants to override the base theme behavior they should specify what they want for the property.
If we implement what @JonFerraiolo is proposing then the user will have no way to get back to the base theme value, they will be forced to override the base theme property with the same value just to get back what cascade was doing for them.

No value is not the same as overriding.

**Comment by jhpedemonte** [comment]
**Clearing** a value is different than **removing the override and defaulting to the base theme**.

If I create a custom theme and then **unset** the `background-image` property, I expect the value of that property to be empty.  So I agree with Jon: unsetting a property on a delta theme shouldn't just remove that property from the delta theme, it should set it to **empty**.

It sounds like we also need a `default` option for properties -- when the user selects this, the code removes the property from the delta theme, meaning that the property from the base theme (the default) is used instead.  And the Properties view needs to be updated to restore the default value in the form element.

**Comment by billreed63** [comment]
I do not understand set it to "empty"

I will use the mobile heading widget  setting the background-image to explain how the code currently works...

The base theme has:

```
.mblHeading {
  position: relative;
  margin: 0;
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  z-index: 1;
  padding: 0;
  height: 42px;
  font-family: Helvetica;
  font-size: 20px;
  font-weight: bold;
  text-align: center;
  line-height: 44px;
  background-image: -webkit-gradient(linear, left top, left bottom, from(#f6f6f6), to(#dedede));
  border-top: 1px solid #bfbfbf;
  border-bottom: 1px solid #bfbfbf;
  color: black;
  text-shadow: rgba(255, 255, 255, 0.5) 0 1px 0;
}
```

The user creates a mycustom theme, they want to use a background color instead: To override the base theme the user must set background-image to "none" the delta theme css will get the following rule added

```
.mblHeading {
    background-image : none;
}
```

The user then decides they like the base theme look for the heading so the want to remove their override. They "empty" the background-image property edit box, the value gets removed from the delta css and the base theme rule is used, restoring the default value for background-image
 The delta css rule looks like:

```
.mblHeading {

}
```

If we do as @JonFerraiolo  proposes making "" equal "none" then for the user to get back to the default behavior the user will have to reenter the   the background-image values that are in the base theme, the theme editor does not display the cascade so the user will have to create a second theme and copy the values over just to restore the default rule. Not very user friendly.

I still believe the code has the correct behavior.

**Comment by JonFerraiolo** [comment]
I need to think about this some, but my inclination right now is that @billreed63 is correct in terms of how the Maqetta authoring model should align with the CSS delta themes model, but I want to stew on this some more to think through the UI issues. So, reassigning to me to think things through, then probably either close the issue, postpone the issue to a future release, or close this issue and create a new issue about UI changes.

**Comment by jhpedemonte** [comment]
You're arguing implementation details.  The user doesn't care that we are using delta themes.

For `background-image`, if I set it to **empty**, I expect that value not to be used any more.  However, under the covers, due to the use of delta themes, what really happens is that we instead use the value as specified in the **base theme**.  This is wrong because it is **not what the user expects**.

For that reason, I proposed adding another item to the dropdown there, specifically to handle delta themes, wherein:
- `none`: sets the rules `background-image: none` in the delta theme
- `default`: unsets any `background-image` rule, thereby defaulting to the value in the base theme

Completely **remove the empty item**, since it doesn't make sense in the world of delta themes.

**Comment by billreed63** [comment]
@jhpedemonte @JonFerraiolo 
-> M8

**Comment by jhpedemonte** [comment]
I'm not sure about moving this to M8.  This seems like a serious issue with the _delta themes_ implementation.

How hard would it be to implement [my proposal](https://github.com/maqetta/maqetta/issues/2143#issuecomment-7650462)?

**Comment by billreed63** [comment]
I disagree with you assumption of what the user expects, If I unset a value I expect it to go back to the base theme value, that is why I "cloned" that theme.

But if we where to implement your proposal we would need to do what  @JonFerraiolo  [described in his comment above](https://github.com/maqetta/maqetta/issues/2143#issuecomment-5776196)

**Comment by JonFerraiolo** [comment]
-> M8 Prio-Med
